collect_app/src/androidTest/java/org/odk/collect/android/support/TestScheduler.java

```diff
@@ -34,12 +34,7 @@ public TestScheduler() {
 
     @Override
     public Cancellable repeat(@NotNull Runnable foreground, long repeatPeriod) {
-        increment();
-
-        return wrappedScheduler.repeat(() -> {
-            foreground.run();
-            decrement();
-        }, repeatPeriod);
+        return wrappedScheduler.repeat(foreground::run, repeatPeriod);
```

Comment:
There's no reason to increment and decrement here as we don't want to idle on repeated foreground work.